pyshtools/shclasses/shcoeffs.py

```diff
@@ -4185,7 +4204,7 @@ def _rotate(self, angles, dj_matrix):
         if dj_matrix is None:
             dj_matrix = _shtools.djpi2(self.lmax + 1)
 
-        cgrid = self.expand(grid='DH')
+        cgrid = self.expand(grid='DH',extend=False)
         rgrid, igrid = cgrid.data.real, cgrid.data.imag
         rgridcoeffs = _shtools.SHExpandDH(rgrid, norm=1, sampling=1, csphase=1)
```

Comment:
For consistency with line 4218 below, we should probably change `_shtools` to `backend()`. Otherwise, if we keep it like this (because there really is not choice, this is the code block for using shtools only), then we should modify line 4218 to use `_shtools` as well.